sway/desktop/fx_renderer/fx_renderer.c

```diff
@@ -879,12 +889,18 @@ void fx_render_blur(struct fx_renderer *renderer, const float matrix[static 9],
 	glUniformMatrix3fv(shader->proj, 1, GL_FALSE, gl_matrix);
 
 	glUniform1i(shader->tex, 0);
-	glUniform1f(shader->radius, blur_radius);
 
 	if (shader == &renderer->shaders.blur1) {
+		glUniform1f(shader->radius, blur_radius);
 		glUniform2f(shader->halfpixel, 0.5f / (renderer->viewport_width / 2.0f), 0.5f / (renderer->viewport_height / 2.0f));
-	} else {
+	} else if (shader == &renderer->shaders.blur2) {
+		glUniform1f(shader->radius, blur_radius);
 		glUniform2f(shader->halfpixel, 0.5f / (renderer->viewport_width * 2.0f), 0.5f / (renderer->viewport_height * 2.0f));
+	} else {
+		glUniform1f(shader->noise, blur_noise);
+		glUniform1f(shader->brightness, blur_brightness);
+		glUniform1f(shader->contrast, blur_contrast);
+		glUniform1f(shader->saturation, blur_saturation);
 	}
```

Comment:
I think these should live in a new shader struct and render method, `blur_effects`. My reasoning is that that blur1 and blur2 are very similar, they perform the same general effect just in a different direction, and they have the same glUniforms. This is more of an effects pass that doesnt have much functionally similar to blur1 and blur2, I don't want potential contributors to see that the blur shader struct has `noise`, `brightness` etc and think that this is a parameter common between all the blur shaders.

Reply:
Amended 8ed8b19

There may be some things being bound that are not needed in `fx_render_blur_effects`, if you could double check there :pray: